cypher-shell/src/main/java/org/neo4j/shell/CommandReader.java

```diff
@@ -0,0 +1,41 @@
+package org.neo4j.shell;
+
+import jline.console.ConsoleReader;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandReader {
+    private final ConsoleReader reader;
+    private final Shell shell;
+    static final Pattern MULTILINE_BREAK = Pattern.compile("\\\\\\s*$");
```

Comment:
mostly my confusion come from a fact that i was thinking that you want to catch '\\   '. And yes you are wright, it should looks like it is now because of ton of escaping. (You can usually use \Q and \E sections in such cases but it also looks strange: '\\Q\\\\E\\s*$' would prefer current variant). Thx!